src/gsy_e/models/strategy/energy_parameters/smart_meter.py

```diff
@@ -38,6 +38,13 @@ def activate(self, area):
         self._energy_profile.read_or_rotate_profiles()
         self._simulation_start_timestamp = area.now
 
+    def decrement_energy_requirement(self, energy_kWh: float, time_slot: DateTime, area_name: str):
```

Comment:
This is an exact duplicate of the one defined in `LoadHoursEnergyParameters`. I decided to keep them separate for the moment because the two classes don't have a common interface/parent and I didn't want to complicate this PR.